test_util/test_upgrade_vpc.py

```diff
@@ -173,9 +172,19 @@ def create_marathon_dns_app(app_id: str, healthcheck_app_id: str) -> dict:
     # DNS resolution app to make sure DNS is available during the upgrade.
     # Periodically resolves the healthcheck app's domain name and logs whether
     # it succeeded to a file in the Mesos sandbox.
+
+    dns_parsed_app_name = marathon_app_id_to_mesos_dns_subdomain(healthcheck_app_id)
+    resolve_name = '{}.marathon.agentip.dcos.thisdcos.directory'.format(dns_parsed_app_name)
+
     return {
         "id": '/' + app_id,
         "cmd": """
+until host -W $TIMEOUT_SECONDS $RESOLVE_NAME
+do
+    echo "Waiting to resolve $RESOLVE_NAME for the first time..."
+    sleep 1
+done
```

Comment:
Previously the wait mechanism was query the mesos dns http api (see the code i deleted). Since this will contain the incorrect info, we need the logic to be in the task